application_manager/plugins/spark_mesos/plugin.py

```diff
@@ -0,0 +1,260 @@
+# Copyright (c) 2017 UFCG-LSD and UPV-GRyCAP.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from application_manager.plugins import base
+from application_manager.service import api
+from application_manager.utils import monitor
+from application_manager.utils import scaler
+from application_manager.utils import mesos
+from application_manager.utils import ssh
+from application_manager.utils.logger import Log, configure_logging
+from application_manager.utils.ids import ID_Generator
+from application_manager.plugins.base import GenericApplicationExecutor
+
+from uuid import uuid4
+
+import json
+import paramiko
+import time
+import threading
+
+plugin_log = Log("Spark-Mesos_Plugin", "logs/mesos_plugin.log")
+configure_logging()
+
+
+class SparkMesosApplicationExecutor(GenericApplicationExecutor):
+
+    def __init__(self, app_id, frameworks_url):
+        self.application_state = "None"
+        self.state_lock = threading.RLock()
+        self.application_time = -1
+        self.start_time = -1
+        self.app_id = app_id
+        self.frameworks_url = frameworks_url
+
+    def get_application_state(self):
+        with self.state_lock:
+            state = self.application_state
+        return state
+
+    def update_application_state(self, state):
+        with self.state_lock:
+            self.application_state = state
+
+    def get_application_execution_time(self):
+        return self.application_time
+
+    def get_application_start_time(self):
+        return self.start_time
+
+    def start_application(self, data):
+        try:
+            self.update_application_state("Running")
+            plugin_log.log("%s | Starting application execution" %
+                          (time.strftime("%H:%M:%S")))
+
+            binary_url = str(data['binary_url'])
+            execution_class = str(data['execution_class'])
+            execution_parameters = str(data['execution_parameters'])
+            expected_time = int(data['expected_time'])
+            number_of_jobs = int(data['number_of_jobs'])
+            actuator = str(data['actuator'])
+            starting_cap = 100 if data['starting_cap'] == "" \
```

Comment:
I don't think this check makes any sense. You should remove it.

Reply:
Removed.

Reply:
Removed.